File: AF/2.py
class Solution:
    def addTwoNumbers(self,l1,l2):
    	ans = ListNode(0)
    	r=ans
    	while(l1 or l2):
            if(l1!=None):
                a=l1.val
            else:a=0
            if(l2!=None):
                b=l2.val
            else:
                b=0
            mid=a+b+ans.val

            if mid>=10:
            	ans.val=mid%10
            	ans.next=ListNode(1)
            else:
            	ans.val=mid
            	ans.next=ListNode(0)
            if(l1!=None):
            	l1=l1.next
            if(l2!=None):
            	l2=l2.next
            if(l1!=None or ans.next.val!=0 or l2!=None):
            	ans=ans.next
            elif(l1==None and l2==None):
            	ans.next=None
    	return r

File: zkf/2.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode output = null;
        ListNode currentNode = null;
        int tempSum = 0;
        int sum = 0;
        int carry = 0;
            
        while (l1 != null || l2 != null) {
            if (l1 == null) {
                tempSum = l2.val + carry;
                sum = tempSum % 10;
                carry = tempSum / 10;
                l2 = l2.next;
            }
            else if (l2 == null) {
                tempSum = l1.val + carry;
                sum = tempSum % 10;
                carry = tempSum / 10;
                l1 = l1.next;
            }
            else {
                tempSum = l1.val + l2.val + carry;
                sum = tempSum % 10;
                carry = tempSum / 10;
                l1 = l1.next;
                l2 = l2.next;
            }
            
            if (currentNode == null) {
                currentNode = new ListNode(sum);
                currentNode.next = null;
                output = currentNode;
            }
            else{
                currentNode.next = new ListNode(sum);
                currentNode = currentNode.next;
                currentNode.next = null;
            }
        }
        if (carry != 0) {
            currentNode.next = new ListNode(carry);
            currentNode = currentNode.next;
            currentNode.next = null;
        }
        return output;
    }
}